### PR TITLE
Torrent download support

### PR DESCRIPTION
**Pull request by wjt**
https://phabricator.endlessm.com/T14035

### CODE REVIEW

File: src/endless/EndlessUsbToolDlg.cpp [hunk 1]
```diff
@@ -20,6 +20,11 @@
 #include "WindowsUsbDefines.h"
 #include "StringHelperMethods.h"
 
+#include "TorrentDownloader.h"
+
+#define min(x, y) (((x) < (y)) ? (x) : (y))
+#define max(x, y)  (((x) > (y)) ? (x) : (y))
```
Comment (wjt) [inline src/endless/EndlessUsbToolDlg.cpp]: This file already uses `min()`, defined in some system header, but something in this branch has broken that. `MIN` and `MAX` are also defined in a Rufus header included by this file, I really don't think we should be re-re-re-defining them.

**Review by wjt — COMMENTED** [review]
It's a bit hard to leave review comments on individual lines because GitHub won't display the diff, so I'm going to have to put them here:

When seeding from a tray icon, I get a lot of “Notification info (tray) untreated message 512” in the log.

 To elaborate on this: on my system, every time I try to build the EndlessUsbTool project, OpenSSL gets re-configured (but not rebuilt, fortunately), and the boost build runs again. It fails because `pyconfig.h` is not on the system. Enough of boost gets built that the rest of the build does ultimately succeed but every successive build of EndlessUsbTool tries and fails to rebuild this. I'd hope that this problem would go away when using a system copy of boost.

I haven't tested the resulting binaries on XP but I noticed the toolchain was switched from vs140_xp to vs140. It's not totally clear to me whether the binaries distributed by Boost are compiled with the correct toolchain to work on XP, so … worth testing this before sinking a load of work into using an out-of-tree boost. Still, I feel like we should be able to use a system-wide copy or at the very least a git submodule.

`TorrentDownloader::NotificationThreadHandler` appears to be a busyloop due to:

``` c++
//std::this_thread::sleep_for(std::chrono::milliseconds(100));
```

I assume this wasn't meant to be commented out. In any case, this loop should use `libtorrent::session::wait_for_alerts` with a timeout, rather than sleeping.

``` c++
    // TODO: find a better way to make sure the torrent library has released the file handle.
    // maybe downloader->m_torrentSession.pause();
    Sleep(3000);
```

Yes, a better way than Sleep(3000) would be nice!

```
+                       // allow user to close app instead of going to tray
+                       m_useLocalFile = true;
```

Overloading this flag to mean "seeding is finished" is a little unpleasant!

**Comment by timofonic** [comment]
@endlessm @wjt Any news about this?

**Comment by wjt** [comment]
No plans to update & land this.

**Pull request closed without merging**